libsrc/c64/emd/c64-65816.s

```diff
@@ -0,0 +1,377 @@
+;
+
+MAP:    sta     curpage                 ; Remember the new page
+        stx     curbank                 ; Remember the new bank
+
+        sta     ptr2+1                  ; src address low
+        lda     #$00
+        sta     ptr2                    ; src address high
+        inx
+        ldy     isnotscpu               ; check if not scpu
+        bne     @notscpu
+        inx
+@notscpu:
+        stx     tmp2                    ; src bank
+
+        sta     tmp1                    ; dst bank
+
+        sta     ptr3+1                  ; length high
+        lda     #$ff
+        sta     ptr3                    ; length low
+
+        lda     #<window
+        sta     ptr1                    ; dst address low
+        ldx     #>window
+        stx     ptr1+1                  ; dst address high
+
+        jsr     transfer
+
+        rts
+
+; ------------------------------------------------------------------------
+; USE: Tell the driver that the window is now associated with a given page.
+
+USE:    sta     curpage                 ; Remember the page
+        stx     curbank                 ; Remember the bank
+        lda     #<window
+        ldx     #>window                ; Return the window
+        rts
+
+; ------------------------------------------------------------------------
+; COMMIT: Commit changes in the memory window to extended storage.
+
+COMMIT: lda     curpage                 ; Get the current page
+        sta     ptr1+1                  ; dst high
+        ldx     #$00
+        stx     ptr1                    ; dst low
+
+        lda     #<window
+        sta     ptr2                    ; src low
+        lda     #>window
+        sta     ptr2+1                  ; src high
+
+        stx     ptr3+1                  ; length high
+        lda     #$ff
+        sta     ptr3                    ; length low
+
+        stx     tmp2                    ; src bank
+        ldy     curbank                 ; Get the current bank
+        iny
+        ldx     isnotscpu
+        bne     @notascpu
+        iny
+@notascpu:
+        sty     tmp1                    ; dst bank
+
+        jsr     transfer
+
+        rts               
+
+; ------------------------------------------------------------------------
+; COPYFROM: Copy from extended into linear memory. A pointer to a structure
+; describing the request is passed in a/x.
+; The function must not return anything.
+;
+
+COPYFROM:
+        sta     ptr4
+        stx     ptr4+1                  ; Save the passed em_copy pointer
+                       
+        ldy     #EM_COPY::COUNT+1       ; start at the end of the struct
+        lda     (ptr4),y                ; get high byte of count
+        tax
+        dey
+        lda     (ptr4),y                ; get low byte of count
+        bne     @nodex
+        dex
+@nodex:
+.P816
+        dec
+.P02
+        sta     ptr3                    ; length low
+        txa
+        sta     ptr3+1                  ; length high

```

Comment:
Now that those `eor #$ff` instructions are gone, you can store the X register directly into `ptr3+1`.